lib/oli/delivery/attempts/core.ex

```diff
@@ -590,7 +590,10 @@ defmodule Oli.Delivery.Attempts.Core do
               where: ra.resource_access_id == ^id,
               order_by: ra.attempt_number,
               select: ra,
-              preload: [:revision]
+              preload: [
```

Comment:
My concern with this approach is that we are now preloading all of the nested activity attempts and part
attempts for the other existing callers to this function. Do we have an idea of what the performance impact
of this could be on these other callers?

A safer approach here would be to create another function in this module called
`preload_activity_part_attempts` that takes a list of resource attempts and does he preloading of
part attempt data. That way we can execute it only where it is required to load all of this data.

Reply:
I agree with your concern. I have already made the changes. Thanks!